### PR TITLE
`z_en_test7.c` Documentation Pass

### PR DESCRIPTION
**Pull request by engineer124**
This does a partial documentation pass of `test7.c`, which covers the song of soaring owl warp cutscene and effects. The wings are handled by c_keyframe, so I'll wait for more docs on that to continue. Also, some of the math was hard to follow so some of those sections also remain undocumented. Any thoughts are welcome!

### CODE REVIEW

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c [hunk 1]
```diff
@@ -47,209 +48,193 @@ ActorInit En_Test7_InitVars = {
...
-        arg0->unk_08.x = arg0->unk_08.x;
-        arg0->unk_08.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
-        Math_SinS(sp26);
-        arg0->unk_08.z = arg0->unk_08.z;
+    feather->pos = *pos;
+
+    if (isPosYRand) {
+        feather->pos.x += Math_CosS(yaw) * 0.0f;
+        feather->pos.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
+        feather->pos.z += Math_SinS(yaw) * 0.0f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Test7/z_en_test7.c:96]: oh wtf. How did you even came up with this?

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c [hunk 2]
```diff
@@ -432,184 +419,188 @@ void EnTest7_Destroy(Actor* thisx, PlayState* play) {
...
         play->unk_18844 = true;
     }
 }
 
-void func_80AF1A2C(EnTest7* this, PlayState* play) {
+void EnTest7_WarpCsPart1(EnTest7* this, PlayState* play) {
     Color_RGB8 fogColor = { 64, 0, 0 };
     Color_RGB8 ambientColor = { 220, 220, 255 };
-    f32 envLerp = this->unk_1E54 / 10.0f;
+    f32 envLerp = this->timer / (f32)(10 - 0);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Test7/z_en_test7.c:435]: Why the `10 - 0`?

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 1 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:96]: If you look right below this code, there is
```c
    feather->velocity.x = Math_CosS(yaw) * velocityXZ;
    feather->velocity.y = Rand_ZeroOne();
    feather->velocity.z = Math_SinS(yaw) * velocityXZ;
```
And both `Math_CosS` and `Math_SinS` do nothing except get a return value, so just calling them as it was before meant nothing. So I assumed it was like the code below except `velocityXZ = 0.0f`

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 2 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:435]: If you look at the various lerps in this file, it's always from a ranger of times i.e. 

`envLerp = 1.0f - (sp2C / (f32)(110 - 100));` (from timer 110 to timer 100)
`f32 lerp = 1.0f - (temp / (f32)(100 - 96));` (from timer 100 to timer 96)
`lerp = (this->timer - 79) / (f32)(95 - 79);` (from timer 95 to timer 79)
`lerp = (this->timer - 42) / (f32)(68 - 42);` (from timer 68 to timer 42)
`lerp = (this->timer - 12) / (f32)(30 - 12);` (from timer 30 to timer 12)

and so on. So I just mimicked that style, even though one of the times is at timer 0

File: src/overlays/actors/ovl_En_Go/z_en_go.c [hunk 3]
```diff
@@ -552,9 +552,9 @@ void EnGo_InitSnow(EnGoEffect effect[ENGO_SNOW_EFFECT_COUNT], Vec3f pos) {
         effect->pos.y += 56.0f;
 
         // Generate a +-15 degree rotational velocity
-        effect->rotVelocity.x = (Rand_ZeroOne() - 0.5f) * (f32)0x1554;
-        effect->rotVelocity.y = (Rand_ZeroOne() - 0.5f) * (f32)0x1554;
-        effect->rotVelocity.z = (Rand_ZeroOne() - 0.5f) * (f32)0x1554;
+        effect->angularVelocity.x = (Rand_ZeroOne() - 0.5f) * (f32)0x1554;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Go/z_en_go.c:555]:
```suggestion
        effect->angularVelocity.x = (Rand_ZeroOne() - 0.5f) * 0x1554;
```
Not sure if better, though still out curiosity are the explicit casts needed?

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c [hunk 4]
```diff
@@ -432,184 +419,188 @@ void EnTest7_Destroy(Actor* thisx, PlayState* play) {
...
 
-            this->unk_144 &= ~1;
+            this->flags &= ~OWL_WARP_FLAGS_DRAW_WINGS;
             player->actor.draw = NULL;
         }
-    } else if (this->unk_1E54 >= 87) {
-        func_80AF082C(this, func_80AF1F48);
-        this->unk_144 &= -9;
-        this->unk_148.unk_10 -= 0x2EE0;
+    } else if (this->timer >= 87) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Test7/z_en_test7.c:473]:
```suggestion
    } else if (this->timer > 86) {
```
There is a function below that actually does `this->timer - 86` so seems nice to reference 86 here as well.

File: src/overlays/actors/ovl_En_Go/z_en_go.c (hunk 3 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Go/z_en_go.c]: No, they're not needed. I removed them

**Review by hensldm — APPROVED** [review]

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c [hunk 5]
```diff
@@ -47,209 +48,193 @@ ActorInit En_Test7_InitVars = {
...
+            feather->angularVelocity.y = -feather->angularVelocity.y;
         }
-        arg0->unk_3A = 0;
+        feather->angularVelocity.z = 0;
     }
 }
 
-void func_80AF0C30(EnTest7Struct2* arg0, Vec3f* arg1, s32 arg2) {
-    static s32 D_80AF3410 = 0;
+void EnTest7_RequestFeather(OwlWarpFeather* feathers, Vec3f* pos, s32 isPosYRand) {
```
Comment (tom-overton) [inline src/overlays/actors/ovl_En_Test7/z_en_test7.c:137]: Maybe this is just me, but when I see the word "request," my first thought is that the function can potentially fail to whatever it's requesting. In this case, though, this function can't fail to add a feather; if it doesn't add it in the loop, it will literally overwrite an existing feather.

I'm trying to think of a replacement for this name, though, and I'm struggling because `AddFeather` is already taken. Maybe the existing `AddFeather` can be `InitFeather`, and then this can be `AddFeather`? Though then `InitFeather` is almost identical to `InitFeathers`, so it's still not great. What do you think?

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c [hunk 6]
```diff
@@ -432,184 +419,188 @@ void EnTest7_Destroy(Actor* thisx, PlayState* play) {
...
-        func_80AF082C(this, func_80AF1CA0);
-        this->unk_144 |= 0x20;
+        EnTest7_SetupAction(this, EnTest7_WarpCsPart2);
+        this->flags |= OWL_WARP_FLAGS_20;
         Audio_PlaySfx_AtPos(&this->actor.projectedPos, NA_SE_PL_WARP_WING_OPEN);
         Play_EnableMotionBlur(120);
     }
 }
 
 void func_80AF1B68(EnTest7* this, PlayState* play) {
```
Comment (tom-overton) [inline src/overlays/actors/ovl_En_Test7/z_en_test7.c:456]: Hmmm, for naming this one, it seems like it's big thing is scaling up the wind capsule and (eventually) hiding the wings. Maybe just `EnTest7_ScaleUpWindCapsule`? Or `EnTest7_GrowWindCapsule`? Something like that.

**Review by tom-overton — COMMENTED** [review]
Nice work, just a few thoughts

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 5 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:137]: That's true, this is a tricky one. When it comes to feathers or effects in general, what's you're opinion on `AddEffect` vs `InitEffect`? Once I'm done the repo sweep of anim cleanups, I want to do something similar for effects and make things consistent like `Add/Init` for example. 

This function is more like `EnTest7_AddAndChooseFeather` since this function primarily chooses which feather index to use, then calls `EnTest7_AddFeather` which is your more typical `AddEffect`/`InitEffect`

File: src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.h [hunk 7]
```diff
@@ -24,7 +24,7 @@ typedef struct ObjKendoKanban {
     /* 0x2FC */ s32 indexLastRootCornerPos;
     /* 0x300 */ s16 hasNewRootCornerPos;
     /* 0x302 */ s16 rotAngle;
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.h:26]: The difference between this and other changes to `rot` is that this is `s16` so it's only 1 specific angle (which one I'm not sure unless I dig more into the code), while the other changes were for `Vec3s` which fits to be the more general `rot`

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 6 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:456]: It does seem like this is the main function to grow the wind capsule, but since it does other things like spinning it, changing flags, and setting up another action func, what would you think of:
`EnTest7_UpdateGrowingWindCapsule`

File: src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.h (hunk 7 above)
[the remark this answers is not in the document]
Comment (tom-overton) [reply src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.h:26]: Ah, I see. Thanks for explaining!

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 5 above)
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:137]: I think I like `InitEffect`, because we tend to use `UpdateEffects` for the function that updates them, and keeping the symmetry with actors is nice to me. As for your second point, `AddAndChooseFeather` is fine with me for this one.

File: src/overlays/actors/ovl_En_Test7/z_en_test7.c (hunk 6 above)
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Test7/z_en_test7.c:456]: Yeah, I think that's fine.

**Review by tom-overton — APPROVED** [review]
